src/data/finset/lattice.lean

```diff
@@ -70,17 +73,24 @@ sup_le (λ b hb, le_trans (h b hb) (le_sup hb))
 lemma sup_mono (h : s₁ ⊆ s₂) : s₁.sup f ≤ s₂.sup f :=
 sup_le $ assume b hb, le_sup (h hb)
 
-@[simp] lemma sup_lt_iff [is_total α (≤)] {a : α} (ha : ⊥ < a) :
-  s.sup f < a ↔ (∀b ∈ s, f b < a) :=
-by letI := classical.dec_eq β; from
-⟨ λh b hb, lt_of_le_of_lt (le_sup hb) h,
-  finset.induction_on s (by simp [ha]) (by simp {contextual := tt}) ⟩
+@[simp] lemma sup_lt_iff [is_total α (≤)] {a : α} (ha : ⊥ < a) : s.sup f < a ↔ (∀ b ∈ s, f b < a) :=
+iff.intro (λ hs b hb, lt_of_le_of_lt (le_sup hb) hs)
+begin
+  intro hs,
+  induction s using finset.induction' with c s hc ih,
+  { exact ha, },
+  { rw [sup_cons, _root_.sup_lt_iff],
+    exact ⟨hs c (mem_cons.2 (or.inl rfl)), ih (λ b h, hs b (mem_cons.2 (or.inr h)))⟩, },
+end
```

Comment:
What's the advantage of this new proof over the old shorter one?

Reply:
As I said above, this is a more constructive proof.

Reply:
Also if every use of `simp` warms the planet by 𝜀°C, then that's 2𝜀°C off right there.